index_alias_impl.go

```diff
@@ -866,3 +856,41 @@ func (f *indexAliasImplFieldDict) Close() error {
 	defer f.index.mutex.RUnlock()
 	return f.fieldDict.Close()
 }
+
+func finalizePreSearchResult(req *SearchRequest, preSearchResult *SearchResult) {
+	if requestHasKNN(req) {
+		preSearchResult.Hits = finalizeKNNResults(req, preSearchResult.Hits)
+	}
+}
+
+// A PreSearchResultProcessor processes the data in
+// the presearch result from multiple
+// indexes in an alias and merges them together to
+// create the final presearch result
+type PreSearchResultProcessor interface {
+	// Add adds the presearch result to the processor
+	Add(*SearchResult, string)
+	// Update the final search result with the finalized
+	// data from the processor
+	Finalize(*SearchResult)
+}
+
+type KnnPreSearchResultProcessor struct {
+	add      func(sr *SearchResult, indexName string)
+	finalize func(sr *SearchResult)
+}
```

Comment:
Again, this feels like we're unnecessarily overcomplicating things.

Drop the `add` and `finalize` function pointers and declare what you need within it's member methods, rather than this redirects below.

Reply:
Ok i take this back - it seems it's because of the the knnCollector in a file within the go build `vectors` flag.